src/sdk/namespace_s3.js

```diff
@@ -240,51 +247,46 @@ class NamespaceS3 {
         dbg.log0('NamespaceS3.read_object_stream:', this.bucket, inspect(_.omit(params, 'object_md.ns')));
         await this._prepare_sts_client();
 
-        return new Promise((resolve, reject) => {
-            /** @type {AWS.S3.HeadObjectRequest & AWS.S3.GetObjectRequest | AWS.S3.CopyObjectRequest} */
-            const request = {
-                Bucket: this.bucket,
-                Key: params.key,
-                Range: params.end ? `bytes=${params.start}-${params.end - 1}` : undefined,
-                PartNumber: params.part_number,
-            };
-            this._set_md_conditions(params, request);
-            this._assign_encryption_to_request(params, request);
-            const req = this.s3.getObject(request)
-                .on('error', err => {
-                    this._translate_error_code(params, err);
-                    dbg.warn('NamespaceS3.read_object_stream:', inspect(err));
-                    reject(err);
-                })
-                .on('httpHeaders', (statusCode, headers, res) => {
-                    dbg.log0('NamespaceS3.read_object_stream:',
+        /** @type {import("@aws-sdk/client-s3").HeadObjectRequest & import("@aws-sdk/client-s3").GetObjectRequest | import("@aws-sdk/client-s3").CopyObjectRequest} */
+        const request = {
+            Bucket: this.bucket,
+            Key: params.key,
+            Range: params.end ? `bytes=${params.start}-${params.end - 1}` : undefined,
+            PartNumber: params.part_number,
+        };
+        this._set_md_conditions(params, request);
+        this._assign_encryption_to_request(params, request);
+        try {
```

Comment:
@naveenpaul1 did you validate this flow? Does it work as expected and update the counter?